python/Ganga/GPIDev/Base/Proxy.py

```diff
@@ -133,22 +147,36 @@ def runtimeEvalString(this_obj, attr_name, val):
 
 
 def raw_eval(val):
+    ## Attempts to evaluate the val object and return the object it evaluates to if it is a Python object
+    ## Makes use of basic caching as we don't expect that things at this level should change.
+    from copy import deepcopy
+    if val in _eval_cache:
+        return deepcopy(_eval_cache[val])
+
     try:
+        ## Catch common strings which shouldn't be evaluated anyway and return them asif an exception occured
+        if val in ['', '.', 'new']:
+            _eval_cache[val] = val
+            return deepcopy(val)
```

Comment:
In the case of `File.name`, that's a `SimpleItem` and so it should only ever be simple Python objects, not `GangaObject`s. This means that `''` should only ever be a string.

There's a bunch of things I'd, like to fix in the schema handling, for example getting rid of the "typelist inferred from type of defvalue" thing. As you say, this PR isn't the place to fix it.